tests/integration/testlib/ovslib.py

```diff
@@ -0,0 +1,113 @@
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+from contextlib import contextmanager
+
+import libnmstate
+from libnmstate.schema import Interface
+from libnmstate.schema import InterfaceState
+from libnmstate.schema import InterfaceType
+from libnmstate.schema import OVSBridge
+
+
+@contextmanager
+def ovs_bridge(name, bridge_subtree_state=None, extra_iface_state=None):
+    desired_state = {
+        Interface.KEY: [
+            {
+                Interface.NAME: name,
+                Interface.TYPE: InterfaceType.OVS_BRIDGE,
+                Interface.STATE: InterfaceState.UP,
+                OVSBridge.CONFIG_SUBTREE: bridge_subtree_state or {},
+            }
+        ]
+    }
+    if extra_iface_state:
+        desired_state[Interface.KEY][0].update(extra_iface_state)
+
+    libnmstate.apply(desired_state, verify_change=False)
+
+    try:
+        yield desired_state
+    finally:
+        libnmstate.apply(
+            {
+                Interface.KEY: [
+                    {
+                        Interface.NAME: name,
+                        Interface.TYPE: InterfaceType.OVS_BRIDGE,
+                        Interface.STATE: InterfaceState.ABSENT,
+                    }
+                ]
+            },
+            verify_change=False,
+        )
+
+
+def add_port_to_bridge(
+    bridge_subtree_state, port_name, port_type, port_state=None
+):
+    if OVSBridge.PORT_SUBTREE not in bridge_subtree_state:
+        bridge_subtree_state[OVSBridge.PORT_SUBTREE] = []
+
+    if port_state is None:
+        port_state = {}
+
+    port_state[OVSBridge.PORT_NAME] = port_name
+    port_state[OVSBridge.PORT_TYPE] = port_type
+    bridge_subtree_state[OVSBridge.PORT_SUBTREE].append(port_state)
+
+    return bridge_subtree_state
+
+
+def create_bridge_subtree_state(options_state=None):
+    if options_state is not None:
+        return {OVSBridge.OPTIONS_SUBTREE: options_state}
+    return {}
+
+
+@contextmanager
+def ovs_interface(name, extra_iface_state=None):
+    desired_state = {
+        Interface.KEY: [
+            {
+                Interface.NAME: name,
+                Interface.TYPE: InterfaceType.OVS_INTERFACE,
+                Interface.STATE: InterfaceState.UP,
+            }
+        ]
+    }
+    if extra_iface_state:
+        desired_state[Interface.KEY][0].update(extra_iface_state)
+
+    libnmstate.apply(desired_state, verify_change=False)
```

Comment:
Why no verification?

Reply:
In my opinion, the verification should happen in the test, not here. Also, it would fail if the bridge is created only after this.

Reply:
See the other response.

Reply:
Done